guava/src/com/google/common/math/Quantiles.java

```diff
@@ -513,7 +542,11 @@ private static void checkIndex(int index, int scale) {
    * ({@code required}, {@code to}] are greater than or equal to that value. Therefore, the value at
    * {@code required} is the value which would appear at that index in the sorted dataset.
    */
-  private static void selectInPlace(int required, double[] array, int from, int to) {
+  @SuppressWarnings(value = {"lowerbound:assignment.type.incompatible",//(1): lowest possible while `to` > 0, `partitionPoint` value is 1, therefore `to` can't be negative.
```

Comment:
I think `to > 0` is not relevant here.
When entering the loop, `required > from` and `from >= 0`, therefore `required >= 1`.
At the assignment, `partitionPoint >= required`, therefore `partitionPoint - 1 >= 0`.